web-ncer/app/services/SampleRequestService.scala

```diff
@@ -0,0 +1,298 @@
+import org.incal.core.dataaccess.Criterion
+import org.incal.core.dataaccess.Criterion._
+import org.incal.play.{Page, PageOrder}
+import play.api.Configuration
+import play.api.libs.json.{JsNull, JsObject, Json}
+import play.api.libs.ws.WSClient
+import play.api.mvc.MultipartFormData.FilePart
+import reactivemongo.bson.BSONObjectID
+import services.BatchOrderRequestRepoTypes.SampleRequestSettingRepo
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+/**
+ * Data needed for creating a view used for sample request submission
+ *
+ * @param dataViewId The id of an existing view to use as a base
+ * @param tableViewParts Information of the data frame (table) to display
+ * @param dataSetSetting Sample request settings associated with a dataset
+ * @param dataSpaceMetaInfos Meta information
+ * @param elementGridWidth Grid width to use for charts
+ */
+case class ActionFormViewData(
+  dataViewId: BSONObjectID,
+  tableViewParts: Seq[TableViewData],
+  dataSetSetting: DataSetSetting,
+  dataSpaceMetaInfos: Traversable[DataSpaceMetaInfo],
+  elementGridWidth: Int
+)
+
+/**
+ * A REMS catalogue item
+ *
+ * @param id The id of the item
+ * @param name The readable name of the item
+ * @param formId The id associated with the catalogue item
+ */
+case class CatalogueItem(
+  id: Int,
+  name: String,
+  formId: Int
+)
+
+/**
+ * Service providing functionality to submit requests to REMS
+ */
+class SampleRequestService @Inject() (
+  dsaf: DataSetAccessorFactory,
+  sampleRequestSettingRepo: SampleRequestSettingRepo,
+  config: Configuration,
+  ws: WSClient,
+  val dataSpaceService: DataSpaceService
+) extends DataSetViewHelper {
+
+  private val remsUrl = config.getString("rems.url").getOrElse(
+    throw new AdaException("Configuration issue: 'rems.url' was not found in the configuration file.")
+  )
+  private val remsServiceUser = config.getString("rems.serviceUser").getOrElse(
+    throw new AdaException("Configuration issue: 'rems.serviceUser' was not found in the configuration file.")
+  )
+  private val remsUserPrefix = config.getString("rems.userPrefix").getOrElse(
+    throw new AdaException("Configuration issue: 'rems.userPrefix' was not found in the configuration file.")
+  )
+  private val remsMasterApiKey = config.getString("rems.masterApiKey").getOrElse(
+    throw new AdaException("Configuration issue: 'rems.masterApiKey' was not found in the configuration file.")
+  )
+
+  /**
+   * Create a valid CSV which can be attached to an REMS application
+   *
+   * @param dataSetId The data set id to use
+   * @param conditions The filter conditions. If empty no filter is applied
+   * @param fieldNames The field/column names to use. If empty all fields are selected
+   * @param selectedIds The selected row ids to use. If empty all are selected
+   * @return A string representing a valid CSV file.
+   */
+  def createCsv(
+    dataSetId: String,
+    conditions: Seq[FilterCondition],
```

Comment:
`FilterCondition` is an UI concept, which should be handled (converted) at the controller level. Services should use `Criterion` (the server/typed representation) exclusively.